lib/vmpooler/metrics/promstats.rb

```diff
@@ -0,0 +1,307 @@
+          metric_suffixes: { invalid: 'Invalid Pool' },
+          param_labels: %i[poolname]
+        },
+        connect: {
+          mtype: M_COUNTER,
+          docstring: 'vmpooler Connect (to vSphere)',
+          prom_metric_prefix: "#{@metrics_prefix}_connect",
+          metric_suffixes: {
+            open: 'Connect Succeeded',
+            fail: 'Connect Failed'
+          },
+          param_labels: []
+        },
+        migrate_from: {
+          mtype: M_COUNTER,
+          docstring: 'vmpooler Machine Migrated from',
+          prom_metric_prefix: "#{@metrics_prefix}_migrate_from",
+          param_labels: %i[host_name]
+        },
+        migrate_to: {
+          mtype: M_COUNTER,
+          docstring: 'vmpooler Machine Migrated to',
+          prom_metric_prefix: "#{@metrics_prefix}_migrate_to",
+          param_labels: %i[host_name]
+        },
+        ready: {
+          mtype: M_GAUGE,
+          docstring: 'vmpooler Number of Machines in Ready State',
+          prom_metric_prefix: "#{@metrics_prefix}_ready",
+          param_labels: %i[poolname]
+        },
+        running: {
+          mtype: M_GAUGE,
+          docstring: 'vmpooler Number of Machines Running',
+          prom_metric_prefix: "#{@metrics_prefix}_running",
+          param_labels: %i[poolname]
+        },
+        connection_available: {
+          mtype: M_GAUGE,
+          docstring: 'vmpooler Redis Connections Available',
+          prom_metric_prefix: "#{@metrics_prefix}_connection_available",
+          param_labels: %i[type provider]
+        },
+        time_to_ready_state: {
+          mtype: M_HISTOGRAM,
+          buckets: POOLER_TIME_BUCKETS,
+          docstring: 'Time taken for machine to read ready state for pool',
+          prom_metric_prefix: "#{@metrics_prefix}_time_to_ready_state",
+          param_labels: %i[poolname]
+        },
+        migrate: {
+          mtype: M_HISTOGRAM,
+          buckets: POOLER_TIME_BUCKETS,
+          docstring: 'vmpooler Time taken to migrate machine for pool',
+          prom_metric_prefix: "#{@metrics_prefix}_migrate",
+          param_labels: %i[poolname]
+        },
+        clone: {
+          mtype: M_HISTOGRAM,
+          buckets: POOLER_TIME_BUCKETS,
+          docstring: 'vmpooler Time taken to Clone Machine',
+          prom_metric_prefix: "#{@metrics_prefix}_clone",
+          param_labels: %i[poolname]
+        },
+        destroy: {
+          mtype: M_HISTOGRAM,
+          buckets: POOLER_TIME_BUCKETS,
+          docstring: 'vmpooler Time taken to Destroy Machine',
+          prom_metric_prefix: "#{@metrics_prefix}_destroy",
+          param_labels: %i[poolname]
+        },
+        connection_waited: {
+          mtype: M_HISTOGRAM,
+          buckets: REDIS_CONNECT_BUCKETS,
+          docstring: 'vmpooler Redis Connection Wait Time',
+          prom_metric_prefix: "#{@metrics_prefix}_connection_waited",
+          param_labels: %i[type provider]
+        }
+      }
+    end
```

Comment:
Your docstring lines have a lot of capital letters that should be lower case. Please update those.